staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -50,13 +50,18 @@ func init() {
 // stores any such user found onto the provided context for the request. If authentication fails or returns an error
 // the failed handler is used. On success, "Authorization" header is removed from the request and handler
 // is invoked to serve the request.
-func WithAuthentication(handler http.Handler, mapper genericapirequest.RequestContextMapper, auth authenticator.Request, failed http.Handler) http.Handler {
+func WithAuthentication(handler http.Handler, mapper genericapirequest.RequestContextMapper, auth authenticator.Request, failed http.Handler, apiAuds []string) http.Handler {
 	if auth == nil {
 		glog.Warningf("Authentication is disabled")
 		return handler
 	}
 	return genericapirequest.WithRequestContext(
 		http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			if len(apiAuds) > 0 {
+				if ctx, ok := mapper.Get(req); ok {
+					mapper.Update(req, authenticator.WithAudiences(ctx, apiAuds))
```

Comment:
is this what we should be updating, or `req.WithContext(authenticator.WithAudiences(req.Context(), apiAuds))`?

Reply:
separately, I wonder if we even need request context mapper now that we can set context directly on the request object, but that's an issue for another PR

Reply:
Ugh, that is not at all confusing.

Reply:
I'm putting together a PR to excise the requestContextMapper from our codebase... shouldn't be necessary any longer

Reply:
opened https://github.com/kubernetes/kubernetes/pull/62810